File: P32151/Yarusova/lab_02/methods/chord_method.py
def chord_method(equation, a, b, accuracy, max_iterations):
    iterations = 0
    x = (a * equation(b) - b * equation(a)) / (equation(b) - equation(a))
    prev = a
    while (iterations < max_iterations) and (abs(equation(x)) > accuracy) and (abs(b - a) > accuracy) and (abs(x - prev) > accuracy):
        iterations += 1
        if equation(a) * equation(x) < 0:
            b = x
        else:
            a = x
        prev = x
        x = (a * equation(b) - b * equation(a)) / (equation(b) - equation(a))
    return x, iterations

File: P32082/Nenov/lab3/calc-math-lab3/src/main/java/org/example/math/eq_solve/ChordMethod.java
package org.example.math.eq_solve;

import org.example.math.Function;
import org.example.math.results.EquationResult;

public class ChordMethod extends EqSolveMethod {

    public ChordMethod(double accuracy, Function function) {
        super(accuracy, function);
    }

    @Override
    public EquationResult calculate(double a, double b) {
        double prX;
        double x = a;
        int iters = 0;
        do {
            prX = x;
            x = (a*function.calculate(b) - b* function.calculate(a)) / (function.calculate(b)- function.calculate(a));
//            System.out.printf("%d) %.3f | %.3f | %.3f | %.3f %.3f | %.3f | %.3f %n", iters+1, a, b, x, function.calculate(a),
//                    function.calculate(b), function.calculate(x), Math.abs(x - prX));
            iters++;
            if (function.calculate(a) * function.calculate(x) < 0)
                a = x;
            else
                b = x;
        } while (Math.abs(x - prX) > accuracy && Math.abs(a-b) > accuracy && Math.abs(function.calculate(x)) > accuracy);
        return new EquationResult(x, iters, function);
    }
}
